### PR TITLE
Update artifacts fetcher to download artifacts locally using FileSystems

### PR DESCRIPTION
**Pull request by AnandInguva**
fixes: https://github.com/apache/beam/issues/30191

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

**Comment by AnandInguva** [comment]
converting this to draft.

**Comment by AnandInguva** [comment]
https://github.com/apache/beam/actions/runs/7805971680 - IT test run

**Comment by AnandInguva** [comment]
https://github.com/apache/beam/actions/runs/7817704461

**Comment by AnandInguva** [comment]
https://github.com/apache/beam/actions/runs/7819538959/job/21332179529

**Comment by AnandInguva** [comment]
R: @tvalentyn

File: sdks/python/apache_beam/ml/transforms/utils.py [hunk 1]
```diff
@@ -18,18 +18,47 @@
...
+from google.cloud.storage import Client
+from google.cloud.storage import transfer_manager
 
-class ArtifactsFetcher():
+
+def download_artifacts_from_gcs(bucket_name, prefix, local_path):
+  """Downloads artifacts from GCS to the local file system.
+    Args:
+        bucket_name: The name of the GCS bucket to download from.
+        folder_name: The name of the folder to download.
```
Comment (tvalentyn) [inline sdks/python/apache_beam/ml/transforms/utils.py:35]:
```suggestion
        prefix: Prefix of GCS objects to download.
```

File: sdks/python/apache_beam/ml/transforms/utils.py [hunk 2]
```diff
@@ -18,18 +18,47 @@
...
+        local_path: The local path to download the folder to.
+    """
```
Comment (tvalentyn) [inline sdks/python/apache_beam/ml/transforms/utils.py:37]:
```suggestion
  """
```

File: sdks/python/apache_beam/ml/transforms/utils.py [hunk 3]
```diff
@@ -18,18 +18,47 @@
...
+  client = Client()
+  bucket = client.get_bucket(bucket_name)
+  blobs = [blob.name for blob in bucket.list_blobs(prefix=prefix)]
+  _ = transfer_manager.download_many_to_path(
+      bucket, blobs, destination_directory=local_path, max_workers=6)
```
Comment (tvalentyn) [inline sdks/python/apache_beam/ml/transforms/utils.py:42]: default for max_workers is 8, any particular reason to reduce to specify it and reduce to 6?

**Review by tvalentyn — APPROVED** [review]
Thanks! PTAL at lint errors.

**Pull request merged**